Subject: config.sub: accept substrate as an OS name

Wire the substrate* OS family into the bundled config.sub so triples
like i386-unknown-substrate stop bouncing off the unrecognised-OS
check at configure time.

---
diff -urN a/build-aux/config.sub b/build-aux/config.sub
--- a/build-aux/config.sub
+++ b/build-aux/config.sub
@@ -1735,7 +1735,7 @@
 	     | os9* | macos* | osx* | ios* \
 	     | mpw* | magic* | mmixware* | mon960* | lnews* \
 	     | amigaos* | amigados* | msdos* | newsos* | unicos* | aof* \
-	     | aos* | aros* | cloudabi* | sortix* | twizzler* \
+	     | aos* | aros* | cloudabi* | sortix* | substrate* | twizzler* \
 	     | nindy* | vxsim* | vxworks* | ebmon* | hms* | mvs* \
 	     | clix* | riscos* | uniplus* | iris* | isc* | rtu* | xenix* \
 	     | mirbsd* | netbsd* | dicos* | openedition* | ose* \
